src/components/SearchField.tsx

```diff
@@ -1,38 +1,232 @@
-import React, { useState } from 'react';
+import React, { useCallback, useState } from 'react';
+import { useHistory } from 'react-router';
+
+import '../styles/SearchField.scss';
 
 const loupeIcon = '/images/search-icon.svg';
 
+const genres = [
+  'none',
+  'Comedy',
+  'Drama',
+  'Romance',
+  'Mystery',
+  'Thriller',
+  'Animation',
+  'Action',
+  'Adventure',
+  'Crime',
+  'Fantasy',
+  'Science Fiction',
+  'Horror',
+  'Family'
+];
+
+const checkedImg = '/images/searchFieldIcons/checked.svg';
+const uncheckedImg = '/images/searchFieldIcons/unchecked.svg';
+
 const SearchField: React.FC = () => {
+  const history = useHistory();
+
   const [inputValue, setInputValue] = useState<string>('');
 
+  const [focus, setFocus] = useState<boolean>(false);
+
+  const [dropdown, setDropdown] = useState<boolean>(false);
+
+  const [dateRange, setDateRange] = useState<{
+    fromDate: number;
+    toDate: number;
+  }>({
+    fromDate: 1980,
+    toDate: 2021
+  });
+
+  const [checkboxes, setCheckboxes] = useState<{
+    favorites: boolean;
+    history: boolean;
+  }>({
+    favorites: false,
+    history: false
+  });
+
+  const [genreToSearch, setGenreToSearch] = useState<string>('none');
+
   const changeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
     setInputValue(event.target.value);
   };
 
-  const submitHandler = (event: React.SyntheticEvent) => {
-    event.preventDefault();
-    if (inputValue.trim()) {
-      console.log(inputValue);
-    }
-  };
+  const submitHandler = useCallback(
+    (event: React.SyntheticEvent) => {
+      event.preventDefault();
+
+      // for proper using of query params need to refactor routing
```

Comment:
should it be // TODO: point?

Reply:
its already in TODO #37 but not described, thank you)